consensus/istanbul/backend/backend.go

```diff
@@ -527,6 +554,17 @@ func (sb *Backend) Verify(proposal istanbul.Proposal) (time.Duration, error) {
 		return 0, err
 	}
 
+	// Cache the BlockProcessResult
```

Comment:
Instead of clearing the cache here, I think it's better to clear it within `NewChainHead` (basically when a new block has been added to the blockchain).

That way, the `pendingBlockProcessResults ` map will only need to cache results for a single block number.   Note that there still can be multiple processed results in a block number, since there can be different proposals for the same sequence.

Reply:
In the new commits, since we refactor the `BlockProcessResults` from `istanbul/backend` to proposalVerificationStatus in `istanbul/core`, and it's cleared at every new sequence. Then moving it within `NewChainHead` is no longer needed.